tests/integration/targets/ec2_vpc_route_table/tasks/main.yml

```diff
@@ -628,47 +629,344 @@
       - recreate_private_table.changed
       - recreate_private_table.route_table.id != create_public_table.route_table.id
 
-  - name: create a VPC endpoint to test ec2_vpc_route_table ignores it
-    ec2_vpc_endpoint:
-      state: present
-      vpc_id: '{{ vpc.vpc.id }}'
-      service: com.amazonaws.{{ aws_region }}.s3
-      route_table_ids:
-      - '{{ recreate_private_table.route_table.route_table_id }}'
-    register: vpc_endpoint
-  - name: purge routes
+  # ------------------------------------------------------------------------------------------
+
+  - name: Create gateway route table - check_mode
     ec2_vpc_route_table:
-      vpc_id: '{{ vpc.vpc.id }}'
+      vpc_id: "{{ vpc.vpc.id }}"
       tags:
-        Public: 'false'
-        Name: Private route table
+        Public: 'true'
+        Name: Gateway route table
+      gateway_id: "{{ vpc_igw.gateway_id }}"
+    register: create_gateway_table
+    check_mode: yes
+
+  - assert:
+      that:
+        - create_gateway_table is changed
+
+  - name: Create gateway route table
+    ec2_vpc_route_table:
+      vpc_id: "{{ vpc.vpc.id }}"
+      tags:
+        Public: 'true'
+        Name: Gateway route table
+      gateway_id: "{{ vpc_igw.gateway_id }}"
+    register: create_gateway_table
+
+  - assert:
+      that:
+        - create_gateway_table is changed
+        - create_gateway_table.route_table.id.startswith('rtb-')
+        - "'Public' in create_gateway_table.route_table.tags and create_gateway_table.route_table.tags['Public']\
+          \ == 'true'"
+        - create_gateway_table.route_table.routes | length == 1
+        - create_gateway_table.route_table.associations | length == 1
+        - create_gateway_table.route_table.vpc_id == vpc.vpc.id
+        - create_gateway_table.route_table.propagating_vgws | length == 0
+
+  - name: Create gateway route table (idempotence) - check_mode
+    ec2_vpc_route_table:
+      vpc_id: "{{ vpc.vpc.id }}"
+      tags:
+        Public: 'true'
+        Name: Gateway route table
+      gateway_id: "{{ vpc_igw.gateway_id }}"
+    register: create_gateway_table
+    check_mode: yes
+
+  - assert:
+      that:
+        - create_gateway_table is not changed
+
+  - name: Create gateway route table (idempotence)
+    ec2_vpc_route_table:
+      vpc_id: "{{ vpc.vpc.id }}"
+      tags:
+        Public: 'true'
+        Name: Gateway route table
+      gateway_id: "{{ vpc_igw.gateway_id }}"
+    register: create_gateway_table
+
+  - assert:
+      that:
+        - create_gateway_table is not changed
+        - create_gateway_table.route_table.id.startswith('rtb-')
+        - "'Public' in create_gateway_table.route_table.tags and create_gateway_table.route_table.tags['Public']\
+          \ == 'true'"
```

Comment:
If possible I think we can break these assertions into two separate ones?

Reply:
Yes I like that idea a lot ty!